### PR TITLE
Explore Variant_Classification's Impact on TMB: Part 1: Filter vs no Filter

### PR DESCRIPTION
**Pull request by cansavvy**
#### What scientific question is your analysis addressing?
Part 1 of #729 

>A notebook that runs the calculate_tmb.R script with and without --nonsynfilter and creates some plots to get visualizes on how the participant data changes with filters.

>How different are the synonymous vs nonsynonymous mutation counts?

Short answer, not much different at all. 

#### What was your approach?

- Does the filter change a participant's TMB? I did a non-filter filter correlation and scatterplot. 
- Are some histologies affected more than others? I facet wrapped the scatterplot by `short_histology`. 
- Does the filter affect the TCGA-PBTA comparison change? I plotted the TMB no filter/filter and TCGA/PBTA side by side. 
- Plot the TMB plot with no filter data - I plotted "the" tmb plot with the no filter version of the tmb and have imported the filtered version for viewing side by side. 

#### What GitHub issue does your pull request address?
Part 1 of #729 

### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

- How do we feel about the set up in general? 
- Does this adequately address part 1 of the question in #729? "How does the nonsynonymous filter affect TMB"?
- Any plots we'd like to see that aren't included? 

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes. Here's the rendered notebook: 
[explore_nonsynfilter.nb.html.zip](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/5103021/explore_nonsynfilter.nb.html.zip)

#### What types of results are included (e.g., table, figure)?

Plots can be seen here [explore_nonsynfilter.nb.html.zip](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/5109615/explore_nonsynfilter.nb.html.zip)

#### What is your summary of the results?

Overall TMB comparisons do not seem to be affected much by filter of nonsynonymous mutations. 

#### Reproducibility Checklist
- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

I haven't added a README since this is a side analysis. The main information is in the Rmd. It's not a lone standing module so I also didn't add it to the table. But if we would like these items to be added, I can. 

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date. 
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by cansavvy** [comment]
Before you get into a fine detail review, @jashapiro can you give this a "big picture" review and see if it fits what you pictured for part 1 of #729 and your comment that that issue is based on: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/728#pullrequestreview-448164485

File: analyses/snv-callers/explore_variant_classifications/explore_nonsynfilter.Rmd [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+Import the tmb calculations that are not filtered by nonsynonymous classifications.
+
+```{r}
+tmb_pbta_no_filter <- data.table::fread(file.path(
+  "..",
+  "results",
+  "no_filter",
+  "pbta-snv-mutation-tmb-coding.tsv"
+)) %>%
+  # This variable is weird when binding but we don't need it for the plot so we'll just remove it.
```
Comment (jashapiro) [inline analyses/snv-callers/explore_variant_classifications/explore_nonsynfilter.Rmd:71]: tell me more?

Comment (cansavvy) [reply analyses/snv-callers/explore_variant_classifications/explore_nonsynfilter.Rmd:71]: Hmm... It was weird when I used this kind of import in another notebook but now its fine, so I deleted this step.

**Comment by cansavvy** [comment]
> If there were not a strong correlation between coding and (coding + noncoding) as you have shown, I would be very surprised, but if the proportions of coding and noncoding changes were different across different tumor types I would be somewhat less shocked (but still surprised).

I'm all for doing this comparison, but it will require different datasets and different set up than what I have going on in this notebook. So can I propose that if what is here is *okay* then I can start a new notebook/ new PR that looks into non-coding/coding proportions?

**Comment by jashapiro** [comment]
> > If there were not a strong correlation between coding and (coding + noncoding) as you have shown, I would be very surprised, but if the proportions of coding and noncoding changes were different across different tumor types I would be somewhat less shocked (but still surprised).
> 
> I'm all for doing this comparison, but it will require different datasets and different set up than what I have going on in this notebook. So can I propose that if what is here is _okay_ then I can start a new notebook/ new PR that looks into non-coding/coding proportions?

My language is imprecise here, I don't really mean coding/noncoding.  I just mean a violin plot (or similar) of `tmb_filter/tmb_no_filter` separated by disease. I think you have the data you need for that in this notebook.

**Comment by cansavvy** [comment]
> > > If there were not a strong correlation between coding and (coding + noncoding) as you have shown, I would be very surprised, but if the proportions of coding and noncoding changes were different across different tumor types I would be somewhat less shocked (but still surprised).
> > 
> > 
> > I'm all for doing this comparison, but it will require different datasets and different set up than what I have going on in this notebook. So can I propose that if what is here is _okay_ then I can start a new notebook/ new PR that looks into non-coding/coding proportions?
> 
> My language is imprecise here, I don't really mean coding/noncoding. I just mean a violin plot (or similar) of `tmb_filter/tmb_no_filter` separated by disease. I think you have the data you need for that in this notebook.

Whew, okay that is much easier. Will do.

**Pull request merged**